blocks/header/header.js

```diff
@@ -1,94 +1,367 @@
+/* eslint-disable prefer-const */
+/* eslint-disable no-shadow */
+import { getMetadata, decorateIcons, decorateButtons } from '../../scripts/lib-franklin.js';
 /**
  * Toggles all nav sections
  * @param {Element} sections The container element
  * @param {Boolean} expanded Whether the element should be expanded or collapsed
  */
-function toggleAllNavSections(sections, expanded = false) {
-  sections.querySelectorAll('.nav-sections > ul > li').forEach((section) => {
-    section.setAttribute('aria-expanded', expanded);
-  });
-}
-
 /**
  * Toggles the entire nav
  * @param {Element} nav The container element
  * @param {Element} navSections The nav sections within the container element
  * @param {*} forceExpanded Optional param to force nav expand behavior when not null
  */
-function toggleMenu(nav, navSections, forceExpanded = null) {
-  const expanded = forceExpanded !== null ? !forceExpanded : nav.getAttribute('aria-expanded') === 'true';
-  const button = nav.querySelector('.nav-hamburger button');
-  document.body.style.overflowY = (expanded || isDesktop.matches) ? '' : 'hidden';
-  nav.setAttribute('aria-expanded', expanded ? 'false' : 'true');
-  toggleAllNavSections(navSections, expanded || isDesktop.matches ? 'false' : 'true');
-  button.setAttribute('aria-label', expanded ? 'Open navigation' : 'Close navigation');
-  // enable nav dropdown keyboard accessibility
-  const navDrops = navSections.querySelectorAll('.nav-drop');
-  if (isDesktop.matches) {
-    navDrops.forEach((drop) => {
-      if (!drop.hasAttribute('tabindex')) {
-        drop.setAttribute('role', 'button');
-        drop.setAttribute('tabindex', 0);
-        drop.addEventListener('focus', focusNavSection);
+/**
+ * decorates the header, mainly the nav
+ * @param {Element} block The header block element
+ */
+
+function createLoginModal() {
+  const loginModal = document.querySelector('nav > div:nth-child(4)');
+  loginModal.classList.add('login-modal');
+
+  const triangle = document.createElement('div');
+  triangle.className = 'triangle';
+  loginModal.appendChild(triangle);
+
+  const divider = document.createElement('div');
+  divider.className = 'divider';
+  loginModal.appendChild(divider);
+}
+
+function handleLoginClick() {
+  const loginModal = document.querySelector('nav > div:nth-child(4)');
+  loginModal.classList.toggle('show');
+
+  const loginButton = document.querySelector('.nav-sections p:last-child');
+  loginButton.classList.toggle('clicked');
+}
+
+function appendUlToP() {
+  let divs = document.querySelectorAll('.mega-menu > div');
+
+  divs.forEach((div) => {
+    let uls = div.querySelectorAll('ul');
+    uls.forEach((ul) => {
+      let p = ul.previousElementSibling;
+
+      let span = document.createElement('div');
+
+      Array.from(ul.children).forEach((li) => {
+        span.textContent += ` ${li.textContent}`;
+      });
+
+      p.appendChild(span);
+
+      ul.remove();
+    });
+  });
+}
+
+/*
+function createTags(links) {
```

Comment:
@basecode I thought to keep it for the follow up issue for the tags, so that the person who will work on the issue already has something to start with

Reply:
I think tags are used everywhere. the code will probably not remain in `header.js`. I guess you can remove.